File: apps/web/src/code/js/basics/process_stream_stdout.js
import { Sandbox } from 'e2b'

const sandbox = await Sandbox.create({
  template: 'base',
  onStdout: (output) => console.log('sandbox', output.line), // $HighlightLine
})

const proc = await sandbox.process.start({
  cmd: 'echo "Hello World!"',
})
await proc.wait()
// output: sandbox Hello World!

const procWithCustomHandler = await sandbox.process.start({
  cmd: 'echo "Hello World!"',
  onStdout: (data) => console.log('process', data.line), // $HighlightLine
})
await procWithCustomHandler.wait()
// output: process Hello World!

await sandbox.close()

File: apps/web/src/code/python/basics/process_stream_stdout.py
from e2b import Sandbox

sandbox = Sandbox(
    template="base",
    on_stdout=lambda output: print("sandbox", output.line),  # $HighlightLine
)

proc = sandbox.process.start('echo "Hello World!"')
proc.wait()
# output: sandbox Hello World!

proc_with_custom_handler = sandbox.process.start(
    'echo "Hello World!"',
    on_stdout=lambda output: print("process", output.line),  # $HighlightLine
)
proc_with_custom_handler.wait()
# output: process Hello World!

sandbox.close()
